db.js

```diff
@@ -280,7 +321,31 @@ exports.init = function (sbot, dir, config) {
     })
   }
 
+  let indexesReady = Obv()
+  indexesReady.once(updateIndexes)
+  let closing = false
+
+  function checkIndexesReady() {
+    let allOk = private.latestSeq.value !== undefined
+    debug(`private seq: ${private.latestSeq.value}`)
+    for (var index in indexes) {
+      if (indexes[index].seq.value === undefined) allOk = false
+      debug(`${index} seq: ${indexes[index].seq.value}`)
+    }
+
+    if (allOk) indexesReady.set(true)
+    else if (!closing) setTimeout(checkIndexesReady, 250)
```

Comment:
Instead of this setTimeout, what if we set `indexesReady` in `updateIndexes()` when the log.stream ends?

Reply:
Yeah this setTimeout is unfortunate. What it does here is that it needs to make sure that all indexes have loaded their state before really starting, this includes any indexes that was added after db was init'ed. So we can't really to it in updateIndexes. Its the other way around.

Reply:
Ah maybe 'multiserver:listening' can be used